src/rest.cpp

```diff
@@ -181,13 +181,14 @@ static bool rest_headers(const std::any& context,
     headers.reserve(count);
     {
         LOCK(cs_main);
-        tip = ::ChainActive().Tip();
-        const CBlockIndex* pindex = g_chainman.m_blockman.LookupBlockIndex(hash);
-        while (pindex != nullptr && ::ChainActive().Contains(pindex)) {
+        ChainstateManager& chainman = EnsureChainman(context);
```

Comment:
nit: Would be nice to do this before locking cs_main, because I think cs_main should be made a (private) member of chainman in the future, if possible.